src/devices/ovs/nm-device-ovs-port.c

```diff
@@ -139,6 +139,13 @@ del_iface_cb (GError *error, gpointer user_data)
 static void
 release_slave (NMDevice *device, NMDevice *slave, gboolean configure)
 {
+	/* Dissociate the platform link from the device -- openvswitch is going
+	 * to remove the link, not us. We shouldn't care about whatever happen to it.
+	 * In particular, we don't want to alter the NMDevice state if we see the
+	 * link go away, because the device may already have another activation
+	 * enqueued. */
+	nm_device_update_from_platform_link (slave, NULL);
```

Comment:
is this really right? Our `NMDevice` instances are usually associated with the netdev device (i.e. the ifindex). Yes, not always, for example bluetooth, WWan and OVS device can exist without an underlying netdev device. But often have devices an `ip_ifindex` that links to the underlying kernel link. For example, we also expose this on D-Bus (as `IpInterface` property).

This means `We shouldn't care about whatever happen to it.` is not true. We do care, as long as there is a link, various places want to know about this association (e.g. D-Bus properties).

I don't think you should clear the ip-ifindex before platform says the link it's really gone. Also, while we may call `nm_device_update_from_platform_link()` from multiple places, I think it always should pass there the `NMPlatformLink` information that is in the platform cache at that point. We shouldn't lie to `NMDevice` about that.

If you don't want to do something with this link, then don't do it (by setting a flag on the device to not doing it). But don't tell NMDevice that the interface is gone, when it is not.